### PR TITLE
:bug: e2e kind cluster creating error

### PR DESCRIPTION
**Pull request by yanmxa**
## Related issue(s)

Fixes # https://github.com/stolostron/multicluster-global-hub/issues/1031

### CODE REVIEW

File: test/script/util.sh [hunk 1]
```diff
@@ -106,7 +106,7 @@ kind_cluster() {
   dir="${CONFIG_DIR:-$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)}"
   local cluster_name="$1"
   local kubeconfig="$dir/$cluster_name"
-  while [ ! -f "$kubeconfig" ]; do
+  while [ ! -f "$kubeconfig" ] || ! kubectl config get-contexts -o name | grep -wq "$cluster_name"; do
```
Comment (clyang82) [inline test/script/util.sh:109]: Do you mean that we have the case to call `kind_cluster` when the cluster is created already?

Comment (yanmxa) [reply test/script/util.sh:109]: No. When we call the `kind_cluster` to create the cluster, we also want to generate the `kubeconfig` and rename the `context` after the cluster is created. However, even if the cluster is created, They may not generate the `kubeconfig` of renaming the `context`. We must re-create the cluster to ensure they are ready.

Comment (clyang82) [reply test/script/util.sh:109]: Did we have logic to re-create the cluster already?

**Pull request merged**